### PR TITLE
[Feature] Support row delimiter for broker load

### PR DESCRIPTION
**Pull request by miomiocat**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Support row delimiter for broker load

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by miomiocat** [comment]
run starrocks_fe_unittest

**Comment by miomiocat** [comment]
run starrocks_admit_test

**Review by caneGuy — APPROVED** [review]

**Comment by miomiocat** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|6|66.67%|[2265, 2266]|

**Review by Astralidea — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -2259,6 +2260,14 @@ private ColumnSeparator getColumnSeparator(StarRocksParser.StringContext context
         return null;
     }
 
+    private RowDelimiter getRowDelimiter(StarRocksParser.StringContext context) {
+        if (context != null) {
+            String sep = ((StringLiteral) visit(context)).getValue();
+            return new RowDelimiter(sep);
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:2266]: Shell we add POS for more correct error message?

**Review by jaogoy — COMMENTED** [review]
It will change the behavior, so we MUST add corresponding content into the document.

**Comment by miomiocat** [comment]
@Mergify backport branch-2.5

**Comment by miomiocat** [comment]
@mergify backport branch-3.0

**Pull request merged**